components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -323,6 +324,20 @@ private void handleRequest(HttpServletRequest req, HttpServletResponse resp, boo
             }
             sendNotification(errorResp, SAMLSSOConstants.Notification.EXCEPTION_STATUS,
                     SAMLSSOConstants.Notification.EXCEPTION_MESSAGE, null, req, resp);
+        } catch (IdentitySAML2ClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred due to an invalid SAML2 SSO request.", e);
+            }
+            String errorResp = null;
+            try {
+                errorResp = SAMLSSOUtil.buildErrorResponse(
+                        SAMLSSOConstants.StatusCodes.IDENTITY_PROVIDER_ERROR,
```

Comment:
What does this indicate?
(Asking since this is a client error and we are returning idp error)

Reply:
We decided to proceed with returning the response as of a idp error to avoid the cause of backward compatibility issues.

Reply:
If we are going to introduce newly, why are we worried about backward compatibility?